scripts/install

```diff
@@ -329,7 +332,7 @@ chown $synapse_user:root -R /var/lib/matrix-$app
 chown $synapse_user:root -R /var/log/matrix-$app
 chown $synapse_user:root -R /etc/matrix-$app
 chmod u=rwX,g=rX,o= -R /etc/matrix-$app
-chmod 600 /etc/matrix-$app/$domain.signing.key
+chmod 600 /etc/matrix-$app/$server_name.signing.key
 setfacl -R -m user:turnserver:rX  /etc/matrix-$app
 setfacl -R -m user:turnserver:rwX  /var/log/matrix-$app
 
```

Comment:
If we go for the .well-known option, we need to add some scripts line to add a json file (server_name.json) into /var/www/.well-known/$domain/server_name.json 
and add a redirection of https://$domain/.well-known/matrix/server to that server_name.json file into /etc/nginx/conf.d/$domain.d/name_server.conf
something like

```
if [[ $server_name != $domain ]]; then
    cat > /var/www/.well-known/${domain}/server_name.json <<EOF
   {
          "m.server": "$domain"
   }
EOF

  cat > /etc/nginx/conf.d/${domain}.d/name_server.conf <<EOF
  {
    location /.well-known/matrix/server {
        alias /var/www/.well-known/${domain}/server_name.json;
        }
  }
EOF
fi
```
I'm a beginner so if I misunderstood something and my script is incorrect, please don't hesitate to comment for me to learn and for the package to improve.

The interest of going through the .well-known system is that we can then write in Readme that the change in DNS is completely optional and if they don't feel comfortable with changing DNS, they better not do it.

Reply:
I'll see to implement the .well-known support. It's also useful for the custom port case.

But it's important to make this change on an other PR, if you want to start on that.

Reply:
Anyway thanks for your useful informations.

Reply:
You're welcome.
I was thinking that maybe @gnouts would like to implement directly the .well-known system instead of the DNS system and as it's the PR for server_name, I thought it was ok to do it here.
It's only a question of some script lines to add (in install and upgrade scripts) and a modification of Readme to write that DNS is not mandatory anymore...
If an other PR is necessary then  I guess we better wait that @gnouts finish this one and it gets merged for good.

Reply:
I have to say that I'm waiting for this feature with impatience to be able to reinstall my synapse server on my family yunohost server.
Everything is like firstname@lastname.xz except matrix which is @firstname:synapse.lastname.xz and I find that not user friendly at all.
E2E cross-signing and this feature are the two missing features for me right now.

Reply:
Hello all,

I have been working on the .well-known solution and, with the help of synapse dev room on matrix, I managed to make it works.
Those who wants to have a look at my work, it's here : https://github.com/Thatoo/synapse_ynh/blob/Thatoo-choose-server-name/scripts/install
Just after install, without changing anything on DNS, the federation can make the relation between my server_name address and my synapse domain going to https://$server_name/.well-known/matrix/server address. However, it is not yet working properly because I guess the 8448 port is only open for the app synapse for entry coming for $domain adn it is closed for entry coming for $server_name address. Is that correct?

I guess I have to change something about that

```
# Open this port
yunohost firewall allow TCP $synapse_tls_port > /dev/null 2>&1
yunohost firewall allow Both $turnserver_tls_port > /dev/null 2>&1
yunohost firewall allow Both $turnserver_alt_tls_port > /dev/null 2>&1

# Store opened ports
ynh_app_setting_set $app synapse_port $port
ynh_app_setting_set $app synapse_tls_port $synapse_tls_port
ynh_app_setting_set $app turnserver_tls_port $turnserver_tls_port
ynh_app_setting_set $app turnserver_alt_tls_port $turnserver_alt_tls_port
ynh_app_setting_set $app cli_port $cli_port
```

Maybe add something to tell to open for this app on both address, $domain and $server_name.

Reply:
Should I make a PR already for us to work on that?

Reply:
@Thatoo I'm closing this discussion as I think you've made your own branch by now and GitHub block merging if there's unresolved discussion :)